crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -290,6 +298,48 @@ def ha_config_check
             end
           end
         end
+
+        # example inputs:
+        # roles_clusters = {
+        #   "neutron-server": "cluster:cluster1",
+        #   "neutron-network": "cluster:cluster1",
+        #   "database-server": "cluster:cluster2",
+        #   "rabbitmq-server": "cluster:cluster3"
+        # }
+        # clusters_roles = {
+        #   "cluster:cluster1": ["neutron-server", "neutron-network"],
+        #   "cluster:cluster2": ["database-server"],
+        #   "cluster:cluster3": ["rabbitmq-server"]
+        # }
+        deployment_supported =
+          case clusters_roles.length
+          when 0
+            # no clusters, no point complaining as this will be detected by other prechecks
+            true
+
+          when 1
+            # everything on one cluster = no problem
+            true
+
+          when 2
+            # neutron-network in separate cluster
+            true if clusters_roles[roles_clusters["neutron-network"]].length == 1 ||
+                # neutron-network + neutron-server in separate cluster
+                (clusters_roles[roles_clusters["neutron-network"]].length == 2 &&
```

Comment:
My understanding was that the cluster should be dedicated to the network services which means that there can be no other services running there.